### ISSUE TITLE
Add linting to CI for Vyper files

### ISSUE THREAD

**Issue by chriswessels** [yearn/yearn-vaults#16]
Currently have linting for Python and Solidity files, but not Vyper.

Add linting support + hook up to Github workflows.

**Comment by spinoch** [comment]
I'm looking into creating such a tool. Is there any chance for compensation in case it resolves this issue? (Not asking for specific numbers, but would be nice to get something)

**Comment by fubuloubu** [comment]
@spinoch yes! I think I can find some individuals that can put together a bounty if you are willing to help take this on!

**Comment by spinoch** [comment]
Cool. I'll see if I can come up with something by next week

**Comment by fameal** [comment]
Hey @spinoch! It seems that you've done a lot of progress in the PR. Are you still working on it?

**Comment by spinoch** [comment]
Hey, @fameal . In short, no. We need to fix a few issues in the blackadder repo and I haven't had time lately. I still would like to get the issues fixed but I can't really give a time frame atm

**Comment by pandadefi** [comment]
We need someone to build and maintain a linter. The linter should be able to work with multiple versions of vyper

**Comment by fubuloubu** [comment]
> We need someone to build and maintain a linter. The linter should be able to work with multiple versions of vyper

note that @spinoch made a first cut of it here: https://github.com/vyperlang/blackadder

**Comment by AbnerZheng** [comment]
@fubuloubu I would like try to finish the remaining work. I have looked through the PR(https://github.com/yearn/yearn-vaults/pull/52), the remaining work is to (correctly) reformat code for mutability of the method, am I right?

**Comment by fubuloubu** [comment]
> @fubuloubu I would like try to finish the remaining work. I have looked through the PR(#52), the remaining work is to (correctly) reformat code for mutability of the method, am I right?

No, there is updates needed to the tool for it to work

**Comment by AbnerZheng** [comment]
@fubuloubu Yeah,  I have checked the tool(https://github.com/vyperlang/blackadder), currently it is simply based on regexp, should I modified it according to the AST to make a normal formatter? Or maybe fork the black(https://github.com/psf/black), and modified the code according to the grammar rule of Vyper?

**Comment by fubuloubu** [comment]
> @fubuloubu Yeah,  I have checked the tool(https://github.com/vyperlang/blackadder), currently it is simply based on regexp, should I modified it according to the AST to make a normal formatter? Or maybe fork the black(https://github.com/psf/black), and modified the code according to the grammar rule of Vyper?

Will accept PRs to that repo, yes

**Comment by AbnerZheng** [comment]
Cool, I will try it.

**Comment by dbale-arg** [comment]
Is there any advance in creating a linter for vyper ?

**Comment by fubuloubu** [comment]
actually yeah, https://github.com/vyperlang/mamushi got a little further in terms of what it can do
